src/classes/BDI_DataImportService.cls

```diff
@@ -41,16 +41,20 @@ global with sharing class BDI_DataImportService {
     public static final String FM_HELP_TEXT = 'Help Text';
     public static final String FM_DATA_IMPORT_FIELD_MAPPING = 'Data Import Field Mapping';
 
-    private BDI_FieldMapping fieldMapper;
+    public BDI_FieldMapping fieldMapper;
 
     /*******************************************************************************************************
     * @description constructor
     * @param isDryRun whether to run in Dry Run mode or commit records.
     */
     public BDI_DataImportService(Boolean isDryRun, BDI_FieldMapping fieldMapper) {
-        contactService = new BDI_ContactService(this);
         this.isDryRun = isDryRun;
         this.fieldMapper = fieldMapper;
+
+        contactService = new BDI_ContactService(this);
+        if (fieldMapper instanceof BDI_FieldMappingCustomMetadata){
```

Comment:
Thanks for updating this from the text comparison to the name - would still prefer this was handled by a method defined in BDI_FieldMapping in the form of "public Boolean supportsAdditionalObjects()" that could allow the field mapper classes to describe their behavior.